shtdlib.sh

```diff
@@ -513,7 +513,7 @@ function finalize_path {
         setvar=false
     else
         debug 5 "Finalizing path for: ${1}"
-        declare path="${!1}"
+        declare path="${!1-unset}"
```

Comment:
Why would you be calling this function without an argument, we might want to add a check for no argument being specified but I don't see a case where we'd want to use unset?

Reply:
Reverted